Milliner/src/Controller/MillinerController.php

```diff
@@ -37,169 +36,260 @@ public function __construct(MillinerServiceInterface $milliner, LoggerInterface
-            $fedora_response = $this->milliner->createRdf(
-                $drupal_jsonld,
-                $drupal_path,
-                $token
-            );
-            return new Response(
-                $fedora_response->getBody(),
-                $fedora_response->getStatusCode(),
-                $fedora_response->getHeaders()
-            );
-        } catch (\Exception $e) {
-            $this->log->debug("Exception Creating Fedora Resource: ", [
-              'body' => $e->getMessage(),
-              'status' => $e->getCode(),
-            ]);
+        $event = json_decode($request->getContent(), true);
+
+        $uuid = $this->parseUuid($event);
+        if (empty($uuid)) {
             return new Response(
-                $e->getMessage(),
-                $e->getCode()
+                "Could not parse UUID from request body",
+                400
             );
         }
-    }
 
-    public function createBinary(
-        ResponseInterface $drupal_entity,
-        Request $request
-    ) {
-        if ($response = $this->processDrupalResponse($drupal_entity)) {
-            return $response;
+        $jsonld_url = $this->parseJsonldUrl($event);
+        if (empty($jsonld_url)) {
+            return new Response(
+                "Could not parse JSONLD url from request body",
+                400
+            );
         }
 
-        $drupal_binary = $drupal_entity->getBody();
-        $mimetype = $drupal_entity->getHeader("Content-Type");
-        $drupal_path = $request->get('path');
-        $token = $request->headers->get('Authorization');
-
         try {
-            $fedora_response = $this->milliner->createBinary(
-                $drupal_binary,
-                $mimetype,
-                $drupal_path,
+            $response = $this->milliner->saveContent(
+                $uuid,
+                $jsonld_url,
                 $token
             );
+
             return new Response(
-                $fedora_response->getBody(),
-                $fedora_response->getStatusCode(),
-                $fedora_response->getHeaders()
+                $response->getBody(),
+                $response->getStatusCode()
             );
+
         } catch (\Exception $e) {
-            $this->log->debug("Exception Creating Fedora Resource: ", [
-                'body' => $e->getMessage(),
-                'status' => $e->getCode(),
-            ]);
-            return new Response(
-                $e->getMessage(),
-                $e->getCode()
-            );
+            $this->log->error("", ['Exception' => $e]);
+            return new Response($e->getMessage(), $e->getCode());
+        }
+    }
+
+    protected function parseUuid(array $event)
+    {
+        if (!isset($event['object']) || !isset($event['object']['id'])) {
+            return null;
+        }
+
+        $urn = $event['object']['id'];
+        if (preg_match("/urn:uuid:(?<uuid>.*)/", $urn, $matches)) {
+            if (isset($matches['uuid'])) {
```

Comment:
Is this necessary? If the preceding `if (preg_match` is successful, this should always succeed... I think. Did you mean to test for emptiness? `if (!empty($matches['uuid'])) {`

Reply:
I'm probably being overly cautious.

Reply:
You are correct.  If the regex matches, the group is always set in the array, but the results can be empty.

```
ubuntu@claw:/var/www/html/Crayfish/Gemini$ php -a
Interactive mode enabled

php > $str = "urn:uuid:";
php > $regex = "/urn:uuid:(?<uuid>.*)/";
php > echo preg_match($regex, $str, $matches);
1
php > echo isset($matches['uuid']);
1
php > echo empty($matches['uuid']);
1
```